lib/cocoapods/sources_manager.rb

```diff
@@ -41,7 +41,15 @@ def aggregate_for_dependency(dependency)
       #
       def sources(names)
         dirs = names.map { |name| source_dir(name) }
-        dirs.map { |repo| Source.new(repo) }
+        dirs.map do |repo|
+          if Pathname(repo).basename == 'master'
```

Comment:
There should be a method like `source_from_path` which takes care of that logic.